src/main/java/com/cloudbees/jenkins/support/SupportAction.java

```diff
@@ -202,9 +226,29 @@ public void doGenerateAllBundles(StaplerRequest req, StaplerResponse rsp) throws
         if (supportPlugin != null) {
             supportPlugin.setExcludedComponents(remove);
         }
+        prepareBundle(rsp, components);
+    }
+
+    /**
+     * Generates a support bundle with only requested components.
+     * @param components component names separated by comma.
+     * @param rsp
+     * @throws IOException
+     */
+    @RequirePOST
+    public void doGenerateBundle(@QueryParameter("components") String components, StaplerResponse rsp) throws IOException {
```

Comment:
I'm missing some permission check here?

Reply:
no, there is already a global check [here](https://github.com/jenkinsci/support-core-plugin/blob/5312de361286eef67f867a039dd33cc48915d2d4/src/main/java/com/cloudbees/jenkins/support/SupportAction.java#L80)